pingcap/tidb-operator
# Support passing raw toml config for tikv/pd as string type

### What problem does this PR solve? 

Support passing raw toml config for tikv/pd as string type
follow up #3327 

also fix a wrong tag name of tikv:
https://github.com/pingcap/tidb-operator/pull/3342/files#diff-35ce941397445e52d38814cd717d06d4R971

### What is changed and how does it work?
Add a TiKVConfigWraper/PDConfigWraper to handle both types of object or string when unmarshaling JSON


### Check List 

Tests 


 - Manual test (add detailed scripts or steps below)
basically like #3327 


Side effects

 - Breaking backward compatibility

The `string` tag removed in pd_config.go is incompatible, but I think we should not use it.

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support passing raw toml config for tikv/pd as string type
```
